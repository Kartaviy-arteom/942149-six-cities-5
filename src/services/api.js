import axios from "axios";

const BASE_URL = `https://5.react.pages.academy/six-cities`;
const TIMEOUT = 5000;

const HttpCode = {
  UNAUTHORIZED: 401
};

export const createAPI = (onUnauthorized) => {
  const api = axios.create({
    baseURL: BASE_URL,
    timeout: TIMEOUT,
    withCredentials: true,
  });

  const onSuccess = (response) => response;

  const onFail = (error) => {
    const {response} = error;

    if (response.status === HttpCode.UNAUTHORIZED) {
      onUnauthorized();

      throw error;
    }

    throw error;
  };

  api.interceptors.response.use(onSuccess, onFail);

  return api;
};
