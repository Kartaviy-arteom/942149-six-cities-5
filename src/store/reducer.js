import offers from "../mocks/offers";
import cities from "../mocks/cities";
import {ActionType} from "./action";

const initialState = {
  cities,
  activeCity: cities[0],
  offers,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_CITY:
      return Object.assign({}, state, {activeCity: action.payload});
  }

  return state;
};

export {reducer};
