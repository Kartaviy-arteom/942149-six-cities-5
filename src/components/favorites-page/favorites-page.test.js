import React from "react";
import renderer from "react-test-renderer";
import FavoritesPage from "./favorites-page";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import reducer from "../../store/reducers/root-reducer";
import {createAPI} from "../../services/api";
import {createStore, applyMiddleware} from "redux";
import thunk from "redux-thunk";
import {composeWithDevTools} from "redux-devtools-extension";

const api = createAPI(() => {});
const store = createStore(
    reducer,
    composeWithDevTools(
        applyMiddleware(thunk.withExtraArgument(api))
    )
);

const cities = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];
const offers = {
  offersCount: 2,
  Paris: [
    {
      amenities: [`Air conditioning`, `Laptop friendly workspace`],
      bedroomsCount: 3,
      city: `Paris`,
      cityCords: [48.85661, 2.351499],
      cityZoom: 13,
      cords: [48.87961000000001, 2.353499],
      costValue: 332,
      description: `Template`,
      isFavorite: true,
      isPremium: false,
      maxGuestsCount: 1,
      offerId: 1,
      offerZoom: 1,
      owner: {
        avatarPath: `img/avatar.jpg`,
        isSuper: true,
        ownerId: 1,
        ownerName: `Boris`
      },
      photoPaths: [`https://`],
      previewImage: `https://`,
      ratingValue: 1,
      title: `Amazing...`,
      type: `apartment`,
    },
  ]
};

it(`Should FavoritesPage render correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <FavoritesPage
              offers={offers}
              cities={cities}
            />
          </BrowserRouter>
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
