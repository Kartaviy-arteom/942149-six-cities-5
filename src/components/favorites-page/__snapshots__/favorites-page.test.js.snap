// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FavoritesPage render correctly 1`] = `
<div
  className="loading"
>
  Loading ...
</div>
`;
